### PR TITLE
Make `PublicKey` type importable

### LINKED ISSUE #717: Allow import of native composite types
### Issue To Be Solved

PR #713 made native composite types, such as `PublicKey` and `SigningAlgorithm` non-storable.
However, this also made them non-importable.

Allow import of native composite types.

### Suggested Solution

- [ ] Add the notion of "importable types", just like the notion of "exportable types", and allow native composite types (ones without a location) to be imported
- [ ] Ensure native composite types can be imported:
  - [ ] `PublicKey`
  - [ ] `SigningAlgorithm`
  - [ ] `HashAlgorithm`
- [ ] Switch the entry point argument checks (script arguments, transaction arguments) from checking storability to importability

### PR DESCRIPTION
**Pull request by SupunS**
Closes #717

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/account_keys_test.go [hunk 1]
```diff
@@ -193,43 +193,6 @@ func TestRuntimeAccountKeyConstructor(t *testing.T) {
 	assert.Contains(t, err.Error(), "cannot find variable in this scope: `AccountKey`")
 }
 
-func TestRuntimeImportAccountAPITypes(t *testing.T) {
```
Comment (SupunS) [inline runtime/account_keys_test.go:196]: Same scenario is already covered in `TestScriptParameterTypeValidation`. So this is a redundant test case.

File: runtime/convertValues.go [hunk 2]
```diff
@@ -465,3 +500,78 @@ func importCompositeValue(
...
+			if !ok {
+				panic(fmt.Errorf(
+					"cannot import value of type '%s'. invalid value for field '%s': %v",
+					sema.PublicKeyType,
+					fieldName,
+					value,
+				))
+			}
+
+			publicKeyValue = arrayValue
```
Comment (turbolent) [inline runtime/convertValues.go:526]: Do we have to validate the array value itself here too, or is this done before / after?

File: runtime/convertValues.go [hunk 3]
```diff
@@ -465,3 +500,78 @@ func importCompositeValue(
...
+			if !ok {
+				panic(fmt.Errorf(
+					"cannot import value of type '%s'. invalid value for field '%s': %v",
+					sema.PublicKeyType,
+					fieldName,
+					value,
+				))
+			}
+
+			signAlgoValue = compositeValue
```
Comment (turbolent) [inline runtime/convertValues.go:539]: Do we have to validate the composite value itself here too, or is this done before / after?

File: runtime/convertValues.go [hunk 4]
```diff
@@ -465,3 +500,78 @@ func importCompositeValue(
...
+
+		case sema.PublicKeyIsValidField:
+			// 'isValid' field set by the user must be ignored.
+			// This is calculated when creating the public key.
+			return
```
Comment (turbolent) [inline runtime/convertValues.go:544]: 👍

**Review by turbolent — COMMENTED** [review]
Nice! Just have one question regarding the validation of the field values

File: runtime/convertValues.go (hunk 2 above)
Comment (SupunS) [reply runtime/convertValues.go:526]: Yes, that gets validated in downstream, when validating the public-key also through the `ConformsToDynamicType` check.

Added a test here: f047877. Also improved public-key validation to gracefully handle errors.

File: runtime/convertValues.go (hunk 3 above)
Comment (SupunS) [reply runtime/convertValues.go:539]: same for the signature algorithm as well.

**Review by turbolent — APPROVED** [review]
Looks good! I have a follow-up PR in #1002

**Pull request merged**